### ISSUE TITLE
Including offline agents in auto-assignment

### ISSUE THREAD

**Issue by adam-nygate** [chatwoot/chatwoot#5672]
**Is your feature request related to a problem? Please describe.**
Suppose agents are currently offline when a conversation is created and they aren't notified of the creation of every conversation (as this can be overwhelming). In that case, conversations are created and left unassigned, with no one aware that they're in limbo.

**Describe the solution you'd like**
I'd like for there to be an opt-in to assign new conversations to all agents (including those that are offline).

**Describe alternatives you've considered**
Subscribing to all notifications (but as mentioned, this can be overwhelming).

**Comment by pranavrajs** [comment]
@adam-nygate Thanks for the suggestion. We have received a lot of feedback where the conversations were assigned to people who were offline and were never attended in time. Hence this approach. At the moment, it would not assign to any offline agent. 

With that in mind, we can provide an option to set the availability manually instead of using the availability based on whether you are active on the web/mobile. This way you can turn off the assignment/availability if you are not working on the support. Let me know what you think.

**Comment by adam-nygate** [comment]
Thanks @pranavrajs, that makes sense. Setting the availability manually would also be a good solution.

**Comment by intrealm** [comment]
we recently upgraded from 2.5.0 to 2.8.1, 
I could see a difference in behaviour since the upgrade

when 1 or more collaboratoring agents are available (no difference here)
  in 2.5.0 new/opened conversation gets assinged to an available agent
  in 2.8.1 new/opened conversation gets assinged to an available agent

when no collaborating agent is available 
  in 2.5.0 new/opened conversation would still gets assinged to a collaborating agents.
  **in 2.8.1 new/opened conversation stays unassigned**

**Comment by pranavrajs** [comment]
@intrealm Makes sense, this is correct. The underlying idea was that none of the offline agents would be assigned a conversation. The conversations would be assigned in a round-robin fashion among the online agents. 

I believe the right approach here would be to provide the flexibility to set the availability rather than making the assignment logic complex.

**Comment by intrealm** [comment]
@pranavrajs thanks for confirming.

however looking at the documentations @ https://www.chatwoot.com/docs/product/inbox/routing-conversations-round-robin/
it clearly says **"If all the agents are offline, the system would distribute the conversations equally to all agents."**

can somebody confirm 
if this feature was taken away in 2.8.1?
or if it is back in the 2.10 releases?
or maybe the documentation is a bit outdated?

**Comment by pranavrajs** [comment]
@intrealm Thanks for pointing this out. The documentation was outdated, I've updated it now.

**Comment by kilrau** [comment]
+1 to let me choose specific agents for auto-assignment when no agent is online

**Comment by carl0xs** [comment]
+1 to let me choose specific agents for auto-assignment when no agent is online :D
